app/controllers/admin/api/cms/templates_controller.rb

```diff
@@ -132,11 +139,21 @@ def find_template
   end
 
   def cms_template_params
-    params.require(:template).permit(*ALLOWED_PARAMS)
+    params.require(:template).permit(*allowed_type_params)
   end
 
   private
 
+  def allowed_type_params
+    ALLOWED_PARAMS[template_type]
+  end
+
+  def template_type
+    return params[:type].parameterize.to_sym if params[:type].present?
+
+    @template.class.name[5..-1].parameterize.to_sym if @template.present?
```

Comment:
Wouldn't this be the same as
```suggestion
    @template.class.data_tag if @template.present?
```
?

Reply:
It's not exactly the same, for instance, for builtin partials this returns `:'builtin-partial'` while `data_tag` is `:builtin_partial`. I could just use the data tags and change the keys in the `ALLOWED_PARAMS` hash. But I don't think that's correct because then we are using the data tags for something they were not intended for. Data tags are to show in responses, let's use them only for that.

Reply:
> because then we are using the data tags for something they were not intended for. Data tags are to show in responses, let's use them only for that.

Agreed! Then it shouldn't be used here **for sure**.

I was wondering if it'd make sense to use `#type` as keys in here but I don't think it's available for all `CMS::` classes. The motivation is to remove this exclusive logic from the controller in favour of something that makes more sense for the `CMS::` fellas.
Anyways, maybe it doesn't make sense to add yet more code there for the models.

Reply:
The `type` is in fact already being used as keys in the hash, it only computes the key as `@template.class.name[5..-1].parameterize.to_sym` when the `type` param isn't received.